Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import AdSupport
+import Shared
+import Leanplum
+
+private let LeanplumEnvironmentKey = "LeanplumEnvironment"
+private let LeanplumAppIdKey = "LeanplumAppId"
+private let LeanplumKeyKey = "LeanplumKey"
+
+private enum LeanplumEnvironment: String {
+    case development = "development"
+    case production = "production"
+}
+
+private struct LeanplumSettings {
+    var environment: LeanplumEnvironment
+    var appId: String
+    var key: String
+}
+
+class LeanplumIntegration {
+    static let sharedInstance = LeanplumIntegration()
+
+    // Setup
+
+    private var profile: Profile?
+
+    func setup(profile: Profile) {
+        if self.profile != nil {
+            Logger.browserLogger.error("LeanplumIntegration - Already initialized")
+            return
+        }
+
+        self.profile = profile
+
+        guard let settings = getSettings() else {
+            Logger.browserLogger.error("LeanplumIntegration - Could not load settings from Info.plist")
+            return
+        }
+
+        switch settings.environment {
+        case .development:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Development")
+            Leanplum.setDeviceId(ASIdentifierManager.shared().advertisingIdentifier.uuidString)
+            Leanplum.setAppId(settings.appId, withDevelopmentKey: settings.key)
+        case .production:
+            Logger.browserLogger.info("LeanplumIntegration - Setting up for Production")
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable : Any]()
+        userAttributesDict["Focus Installed"] = "false"
+        userAttributesDict["Klar Installed"] = "false"
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+
+        if let focusURL = NSURL(string: "firefox-focus://"), UIApplication.shared.canOpenURL(focusURL as URL) {
+            userAttributesDict["Focus Installed"] = "true"
+        }
+
+        if let klarURL = NSURL(string: "firefox-klar://"), UIApplication.shared.canOpenURL(klarURL as URL) {
+            userAttributesDict["Klar Installed"] = "true"
+        }
+
+        if let mailtoScheme = profile.prefs.stringForKey(PrefsKeys.KeyMailToOption), mailtoScheme != "mailto:" {
+            userAttributesDict["Alternate Mail Client Installed"] = mailtoScheme
+        }
+
+        Leanplum.start(userAttributes: userAttributesDict)
+    }
+
+    // Events
+
+    func track(event: String) {
```

Comment:
How about instead of using a `String` here, use an enum. Then it is more strictly controlled and there will also be  less chance of typos.

We do the same for the Adjust integration in Focus. See:

- https://github.com/mozilla-mobile/focus/blob/master/Blockzilla/AdjustIntegration.swift#L8
- https://github.com/mozilla-mobile/focus/blob/master/Blockzilla/AdjustIntegration.swift#L93